rt/rs/extensions/reactor/src/main/java/org/apache/cxf/jaxrs/reactor/client/ReactorInvokerImpl.java

```diff
@@ -201,13 +202,15 @@
     @Override
     public <T> Flux<T> flux(String name, Entity<?> entity, Class<T> responseType) {
         Future<Response> futureResponse = webClient.async().method(name, entity);
-        return Flux.fromIterable(toIterable(futureResponse, responseType));
+        return Flux.fromStream(() -> 
+            StreamSupport.stream(toIterable(futureResponse, responseType).spliterator(), false));
     }
 
     @Override
     public <T> Flux<T> flux(String name, Class<T> responseType) {
         Future<Response> futureResponse = webClient.async().method(name);
-        return Flux.fromIterable(toIterable(futureResponse, responseType));
+        return Flux.fromStream(() -> 
```

Comment:
So when the `toIterable` fails (fe empty response, or alike), it does thrown the exception from `getFlux` method, which is unexpected. Instead, the exception should be propagated using `onError` callback, following the reactive stream.